models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -121,30 +121,107 @@ public function save(Localizedfield|AbstractData|\Pimcore\Model\DataObject\Objec
         }
 
         $data = $this->getDataFromObjectParam($object, $params);
+        $myCurrentRawRelations = $this->prepareMyCurrentRelations($object, $params);
+        $myNewRawRelations = [];
+        $ignoreClassId = null;
+        $classId = null;
+
+        if ($object instanceof Concrete) {
+            $classId = $object->getClassId();
+        } elseif ($object instanceof AbstractData) {
+            $classId = $object->getObject()->getClassId();
+        } elseif ($object instanceof Localizedfield) {
+            $classId = $object->getObject()->getClassId();
+        } elseif ($object instanceof \Pimcore\Model\DataObject\Objectbrick\Data\AbstractData) {
+            $classId = $object->getObject()->getClassId();
+        } else {
+            throw new \Exception('Invalid object type');
+        }
```

Comment:
What do you think about writing it like this?

```suggestion
        $classId = match (true) {
        	$object instanceof Concrete => $object->getClassId(),
        	$object instanceof AbstractData => $object->getObject()->getClassId(),
        	$object instanceof Localizedfield => $object->getObject()->getClassId(),
        	$object instanceof \Pimcore\Model\DataObject\Objectbrick\Data\AbstractData => $object->getObject()->getClassId(),
        	default => throw new \Exception('Invalid object type'),
        };
```

Reply:
nice